contrib/dockerd-rootless.sh

```diff
@@ -139,6 +160,25 @@ if [ -z "$_DOCKERD_ROOTLESS_CHILD" ]; then
 		"$0" "$@"
 else
 	[ "$_DOCKERD_ROOTLESS_CHILD" = 1 ]
+
+	# The Container Device Interface (CDI) specs can be found by default
+	# under {/etc,/var/run}/cdi. More information at:
+	# https://github.com/cncf-tags/container-device-interface
+	#
+	# In order to use the Container Device Interface (CDI) integration,
+	# the CDI paths need to exist before the Docker daemon is started in
+	# order for it to read the CDI specification files. Otherwise, a
+	# Docker daemon restart will be required for the daemon to discover
+	# them.
+	#
+	# If another set of CDI paths (other than the default /etc/cdi and
+	# /var/run/cdi) are configured through the Docker configuration file
+	# (using "cdi-spec-dirs"), they need to be bind mounted in rootless
+	# mode; otherwise the Docker daemon won't have access to the CDI
+	# specification files.
+	[ -d "/etc/cdi" ] && mount_directory /etc/cdi
```

Comment:
Won't these checks trigger the script to exit early due to `set -e`?

Reply:
I think it's fine because they are in a conditional, checked that it works:

```
#!/bin/sh

set -e

mount_directory() {
    echo "mount_directory $1;"
}

[ -d "/non/existant/path" ] && mount_directory /non/existant/path
[ -d "/etc/cdi" ] && mount_directory /etc/cdi
[ -d "/var/run/cdi" ] && mount_directory /var/run/cdi
```

Yielding:

```
❯ ./test.sh
mount_directory /etc/cdi;
mount_directory /var/run/cdi;
```

Reply:
That's not a conditional, though.
It is the same as `test -d <dir>`, which exits non-zero if the directory does not exist.

```console
$ [ -d /foo ] || echo $?
1
```

In the case of this PR, if any of these tested dirs do not exist, the script will exit early.

Here's a more complete example


```bash
foo() {
  [ -d /does/not/exist ]
  echo "after dir check"
}

set -e

foo
echo "after foo"
```

The placement of `set -e` doesn't matter there so long as we put it before calling `foo`.
If you copy/paste this into your shell (maybe do it in a subshell) it will exit with code `1` and not output anything.

Reply:
Thanks for your explanation. Updated to perform the check in a proper conditional. Hope I didn't overlook anything this time.